packages/components/src/content-typings.ts

```diff
@@ -0,0 +1,162 @@
+export const messageTypes = [
+  'text',
+  'audio',
+  'file',
+  'video',
+  'image',
+  'dropdown',
+  'visit',
+  'voice',
+  'typing',
+  'card',
+  'carousel',
+  'location',
+  'quick_reply',
+  'login_prompt',
+  'session_reset',
+  'custom',
+  'unsupported' // This is a special type that is used as fallback for unsupported message types at runtime
+] as const
+
+type MessageTypeTuple = typeof messageTypes
+
+export type MessageType = MessageTypeTuple[number]
```

Comment:
Wouldn't that work instead?

```suggestion
export type MessageType =
  | 'text'
  | 'audio'
  | 'file'
  | 'video'
  | 'image'
  | 'dropdown'
  | 'visit'
  | 'voice'
  | 'typing'
  | 'card'
  | 'carousel'
  | 'location'
  | 'quick_reply'
  | 'login_prompt'
  | 'session_reset'
  | 'custom'
  | 'unsupported' // This is a special type that is used as fallback for unsupported message types at runtime
```

Reply:
The reason I did it like this is to be able to use the array of content types at runtime, this behaviour is no longer required (except for one test) but I don't see a reason why we wouldn't keep it?

Reply:
Fair enough